### PR TITLE
Added more attributes to unittests so they can catch more regressions.

### PR DESCRIPTION
**Pull request by dukc**
More is more here, no?

### CODE REVIEW

File: std/range/package.d [hunk 1]
```diff
@@ -6269,15 +6269,22 @@ FrontTransversal!(RangeOfRanges, opt) frontTransversal(
...
+pure @safe nothrow unittest
 {
     int[][] arr = [[], []];
 
     auto ft1 = frontTransversal!(TransverseOptions.assumeNotJagged)(arr);
     assert(ft1.empty);
 
-    auto ft2 = frontTransversal!(TransverseOptions.enforceNotJagged)(arr);
-    assert(ft2.empty);
+    try
```
Comment (MetaLang) [inline std/range/package.d]: It's not worth having `nothrow` on the unittest if it makes the example this ugly.

File: std/range/package.d [hunk 2]
```diff
@@ -5006,28 +5007,28 @@ struct Recurrence(alias fun, StateType, size_t stateSize)
...
-@safe unittest
+pure @safe nothrow @nogc unittest
 {
     import std.algorithm.comparison : equal;
 
     // The Fibonacci numbers, using function in string form:
     // a[0] = 1, a[1] = 1, and compute a[n+1] = a[n-1] + a[n]
     auto fib = recurrence!("a[n-1] + a[n-2]")(1, 1);
-    assert(fib.take(10).equal([1, 1, 2, 3, 5, 8, 13, 21, 34, 55]));
+    assert(fib.take(10).equal(only(1, 1, 2, 3, 5, 8, 13, 21, 34, 55)));
```
Comment (JackStouffer) [inline std/range/package.d]: I don't think we should complicate ddoc-ed unit tests

File: std/range/package.d (hunk 1 above)
Comment (MetaLang) [inline std/range/package.d]: I didn't realize this was not a DDOC'd unit test, so it's not as big of a deal, but I still don't like this and don't really see much of a point.

Comment (jmdavis) [reply std/range/package.d]: The test can be split into a separate `unittest` block if it can't be `nothrow`.

File: std/range/package.d (hunk 2 above)
Comment (dukc) [reply std/range/package.d]: I can revert that, but this means I also have to take off the nogc attribute I added.

**Comment by dukc** [comment]
Even that style article itself does not obey that in the other guidelines, lol!

File: std/range/package.d (hunk 2 above)
Comment (JackStouffer) [reply std/range/package.d]: Yes, I believe so. Making the ddoc-ed unit tests simpler is always better IMO. If you want to, you can always add a separate unit test block for the express purpose of testing `@nogc`, e.g. https://github.com/dlang/phobos/blob/9503a1e11ed9c474464b1ea6e5b07391407b18d3/std/range/package.d#L11862 (I would merge this even without that addition though).

**Comment by dukc** [comment]
Autotester stuck. Restarting.

**Pull request merged**